simple/stats/events_importer.py

```diff
@@ -0,0 +1,280 @@
+      self._rename_columns()
+      self._write_event_triples()
+      self._write_observations()
+      self.reporter.report_success()
+    except Exception as e:
+      self.reporter.report_failure(str(e))
+      raise e
+
+    self._write_debug_csvs()
+
+  def _read_csv(self) -> None:
+    # Read CSVs with the following behaviors:
+    # - Set 1st column (i.e. the entity column) to type str (so that geoIds like "01" are not treated as ints and converted to 1)
+    # - Strip leading whitespaces
+    # - Treat comma as a thousands separator
+    self.df = pd.read_csv(self.input_fh.read_string_io(),
+                          dtype={0: str},
+                          skipinitialspace=True,
+                          thousands=",")
+    logging.info("Read %s rows.", self.df.index.size)
+
+  def _drop_ignored_columns(self):
+    if self.ignore_columns:
+      self.df.drop(columns=self.ignore_columns, axis=1, inplace=True)
+
+  def _sanitize_values(self):
+    # Convert to best possible dtypes (i.e. keep ints as ints even when some values are NaN)
+    self.df = self.df.convert_dtypes()
+    # Set date field to type str.
+    self.df = self.df.astype({self.df.columns[1]: str})
+
+  def _rename_columns(self) -> None:
+    renamed = {}
+    # Rename dcid and date columns
+    renamed[self.df.columns[0]] = constants.COLUMN_DCID
+    renamed[self.df.columns[1]] = constants.COLUMN_DATE
+
+    # Rename property columns to their IDs
+    property_column_names = self.df.columns[2:]
+    property_ids = [
+        self.nodes.property(property_column_name).dcid
+        for property_column_name in property_column_names
+    ]
+    renamed.update({
+        col: id for col, id in zip(property_column_names, property_ids)
+    })
+
+    self.df = self.df.rename(columns=renamed)
+
+  def _parse_dates(self) -> None:
+    # TODO: parse dates and update value if needed (e.g. year, month, day).
+    pass
+
+  def _write_observations(self) -> None:
+    # TODO: compute aggregated observations and insert in DB.
+    pass
+
+  def _write_event_triples(self) -> None:
+    # Add event type node - it will be written to DB later.
+    # This is to avoid duplicate event types in scenarios where events of the same type
+    # are spread across files.
+    self.nodes.event_type(self.event_type, self.input_file_name)
+
+    # All property columns would've been renamed to their dcids by now.
+    # So use the id column's dcid as the id column name.
+    id_column_name = self.nodes.property(
+        self.id_column).dcid if self.id_column else ""
+
+    if not self.entity_type:
+      self.entity_type = self._resolve_entity_type()
+      if self.entity_type:
+        logging.info("Resolved entity type: %s", self.entity_type)
+    if not self.entity_type:
+      logging.warning(
+          "Could not resolve entity type. Event triples will not be associated with any entity."
+      )
+
+    triples: list[Triple] = []
+    for index, row in self.df.iterrows():
```

Comment:
prefer to use itertuples than iterrows as it has a bunch of pros (from my recent pandas learnings)

Reply:
I need to access values both by index and name, so using `iterrows`.